src/hstring.rs

```diff
@@ -67,6 +79,29 @@ impl HString {
     }
 }
 
+impl HString {
```

Comment:
Why not just combine this in the previous `impl`?

Reply:
I thought it was best to separate for readability since this method is a private constructor and others are all public. I'll combine the impl blocks.